src/inference/inference_tensorflowlite.py

```diff
@@ -232,31 +204,54 @@ def raw_result_output(average_time, fps, latency):
 
 def create_dict_for_transformer(args):
     dictionary = {}
-    for i, name in enumerate(args.input_name):
-        channel_swap = args.channel_swap[i] if i < len(args.channel_swap) else None
+    for name in args.input_names:
+        channel_swap = args.channel_swap.get(name, None)
         mean = args.mean.get(name, None)
         input_scale = args.input_scale.get(name, None)
-        layout = args.layout[i] if i < len(args.layout) else 'NHWC'
+        layout = args.layout.get(name, 'NHWC')
         dictionary[name] = {'channel_swap': channel_swap, 'mean': mean,
                             'input_scale': input_scale, 'layout': layout}
 
     return dictionary
 
 
-def parse_mean_scale_arg(values, input_names):
+def parse_input_arg(values, input_names):
     return_values = {}
     if values is not None:
         matches = re.findall(r'(.*?)\[(.*?)\],?', values)
         if matches:
             for i, match in enumerate(matches):
                 name, value = match
                 value = ast.literal_eval(value)
+                if name != '':
+                    return_values[name] = list(value)
+                else:
+                    if input_names is None:
+                        raise ValueError('Please set --input-names parameter'
+                                         ' or use input0[value0],input1[value1] format')
+                    return_values[input_names[i]] = list(value)
+        else:
+            raise ValueError(f'Unable to parse input parameter: {values}')
+    return return_values
+
+
+def parse_layout_arg(values, input_names):
+    return_values = {}
+    if values is not None:
+        matches = re.findall(r'(.*?)\((.*?)\),?', values)
+        if matches:
+            for i, match in enumerate(matches):
```

Comment:
`i` is not used so you can iterate directly, without enumerate

Reply:
`i` is used when input names are provided separately in `--input_names` parameter. https://github.com/YADRO-KNS/dl-benchmark/pull/6/files/fc9f43619cd2126a66d6e3f18e79e4bb9b69c5c8#diff-1c1d9f4bce6f85f6e29a0f5d57303e1a0da9e4f956feac651c2a61b431bd6e1aL257

Reply:
Missed, ok!